src/Microsoft.AspNet.Server.Kestrel/ServerFactory.cs

```diff
@@ -43,7 +43,8 @@ public IDisposable Start(IServerInformation serverInformation, Func<IFeatureColl
             {
                 disposables.Add(engine.CreateServer(
                     address.Scheme,
-                    address.Host,
+                    // Unix sockets use a file path, not a hostname.
+                    address.Scheme == Constants.UnixScheme ? address.Path : address.Host,
```

Comment:
@Daniel15 @lodejard 

[Here is a change to ServerAddress](https://github.com/aspnet/KestrelHttpServer/commit/6b0dfda0aea65a00421c5cab6e278f67aa30385b) that supports your suggested URI syntax.

@lodejard Does this look like what you want? If so, I can start adding badly needed URI parsing tests for this.